matRad_calcSigmaIni.m

```diff
@@ -0,0 +1,72 @@
+function [sigmaIni] = matRad_calcSigmaIni(baseData,rays,SSD)
+% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+% This function evaluates simultaneously the initial sigma of the beam for
+% one or more energies
+%
+% call
+%   sigmaIni = matRad_calcSigmaIni(machine.data,stf(i).ray,stf(i).ray(j).SSD);
+%
+% input
+%   baseData:           'machine.data' file
+%   rays:            	'stf.ray' file
+%   SSD:                source-surface difference
+%
+% output
+%   sigmaIni:        	initial sigma of the ray at certain energy (or
+%                       energies). The data is given in 1xP dimensions,
+%                       where 'P' represents the number of different
+%                       energies
+%
+% References
+%
+%
+% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+%
+% Copyright 2017 the matRad development team.
+%
+% This file is not part of the offical matRad release
+%
+% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+% Finds the energies involved in the process and their focus index
+[energyVec,uniqueEnergyIdx] = unique([rays.energy]);
+tempFocus = [rays.focusIx];
+focusIxVec = tempFocus(uniqueEnergyIdx);
+
+% finds energy index for all the energies
+[~,energyIxVec] = intersect([baseData.energy],energyVec);
+
+if length(energyVec)==1
+    sigmaIni = matRad_interp1(baseData(energyIxVec).initFocus.dist(rays.focusIx,:)',baseData(energyIxVec).initFocus.sigma(rays.focusIx,:)',SSD);
+    
+else
+    % finds standard deviation and distance for all the energies
+    focusStruct = [baseData.initFocus];
+    sigmaVec = [focusStruct(energyIxVec).sigma];
+    distVec = [focusStruct(energyIxVec).dist];
+    
+    dim = size(focusStruct(1).sigma,2);
+    
+    % repeats every index for a number of times equal to size of sigma
+    repFocusIxVec = kron(focusIxVec, ones(1,dim));
```

Comment:
I don't know about the efficiency of using kron, wouldn't repmat do the trick here?
Maybe this whole block could be implemented more efficiently using bsxfuns, but I am not sure.

Reply:
repelem does the job perfectly. I'll change the line.